tests/test/projection.rs

```diff
@@ -444,6 +536,7 @@ fn normalize_under_binder() {
             }
         }
 
+        /* TODO: this doesn't seem to be correct, actually
```

Comment:
Again, let's document the behavior of the two solvers -- what does recursive solver do here?

I think I expect... U = I32?